domain/status/modelmigration/export.go

```diff
@@ -103,6 +117,36 @@ func (e *exportOperation) Execute(ctx context.Context, m description.Model) erro
 	return nil
 }
 
+func (e *exportOperation) exportMachineStatuses(
+	ctx context.Context,
+	service ExportService,
+	m description.Model,
+) error {
+	machineStatuses, instanceStatuses, err := service.ExportMachineStatuses(ctx)
+	if err != nil {
+		return errors.Errorf("retrieving machine statuses: %w", err)
+	}
+
+	for _, machine := range m.Machines() {
+		instance := machine.Instance()
+		machineName := coremachine.Name(machine.Id())
+
+		machineStatus, ok := machineStatuses[machineName]
+		if !ok {
+			return errors.Errorf("machine %q has no status", machineName)
+		}
+		instanceStatus, ok := instanceStatuses[machineName]
+		if !ok {
+			return errors.Errorf("machine %q has no instance status", machineName)
+		}
+
+		machine.SetStatus(e.exportStatus(machineStatus))
+		instance.SetStatus(e.exportStatus(instanceStatus))
```

Comment:
Do we not also need to set the status on the containers as well?

Reply:
I will do this in a follow up PR, containers are a little more fiddly